src/tokens/TokenImport.tsx

```diff
@@ -63,13 +68,41 @@ export default function TokenImportScreen(_: Props) {
       if (supportedTokens[tokenId]) {
         setAddressState(AddressState.AlreadySupported)
       } else {
-        setAddressState(AddressState.Valid)
+        validateCheck(address)
       }
     } else {
       setAddressState(AddressState.Invalid)
     }
   }
 
+  const validateCheck = async (address: Address) => {
+    setAddressState(AddressState.FetchingContract)
+
+    const contract = getContract({
+      abi: erc20.abi,
+      address,
+      publicClient: publicClient[networkIdToNetwork[networkId]],
+    })
+
+    const state = await Promise.race<AddressState>([
+      new Promise((resolve) => {
+        Promise.all([
+          contract.read.symbol(),
+          contract.read.decimals(),
+          contract.read.name(),
+          contract.read.totalSupply(),
```

Comment:
do we need total supply? we'll eventually need to also query the user's balance, so could we add a balance check here?